src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java

```diff
@@ -209,43 +260,148 @@
      */
     private static final int HIDDEN_ID_POSITION = 5;
 
+    /**
+     * The distance to move the ribbon.
+     */
+    private static final int RIBBON_OFFSET = 5;
+
     /**
      * Link to the xml file of the window.
      */
     private static final String TOWNHALL_RESOURCE_SUFFIX = ":gui/windowTownHall.xml";
 
     /**
-     * The builders job description string.
+     * The button to go to the previous permission settings page.
+     */
+    private static final String BUTTON_PREV_PAGE_PERM = "prevPagePerm";
+
+    /**
+     * The button to go to the next permission settings page.
+     */
+    private static final String BUTTON_NEXT_PAGE_PERM = "nextPagePerm";
+
+    /**
+     * The button to go to the officer permission settings page.
+     */
+    private static final String BUTTON_MANAGE_OFFICER = "officerPage";
+
+    /**
+     * The button to go to the friend permission settings page.
+     */
+    private static final String BUTTON_MANAGE_FRIEND = "friendPage";
+
+    /**
+     * The button to go to the neutral permission settings page.
+     */
+    private static final String BUTTON_MANAGE_NEUTRAL = "neutralPage";
+
+    /**
+     * The button to go to the hostile permission settings page.
+     */
+    private static final String BUTTON_MANAGE_HOSTILE = "hostilePage";
+
+    /**
+     * Id of the switch view of the perm pages.
+     */
+    private static final String VIEW_PERM_PAGES = "permPages";
+
+    /**
+     * Id of the switch view of the different groups.
+     */
+    private static final String VIEW_PERM_GROUPS = "userGroups";
+
+    /**
+     * The view of the permission management.
+     */
+    private static final String PERMISSION_VIEW = "managePermissions";
+
+    /**
+     * Button to trigger the permission changes.
+     */
+    private static final String BUTTON_TRIGGER = "trigger";
+
+    /**
+     * The id of the officer permission view.
+     */
+    private static final String VIEW_OFFICER = "officer";
+
+    /**
+     * The id of the officer permission view.
+     */
+    private static final String VIEW_FRIEND = "friend";
+
+    /**
+     * The id of the officer permission view.
+     */
+    private static final String VIEW_NEUTRAL = "neutral";
+
+    /**
+     * The id of the officer permission view.
+     */
+    private static final String VIEW_HOSTILE = "hostile";
+
+    /**
+     * The list of actions for a certain permission group.
+     */
+    private static final String LIST_ACTIONS = "actions";
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Use already-defined constant 'BUTTON_ACTIONS' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1192)